### PR TITLE
Remove explicit dependency on 2dotstwice/collection

### PR DESCRIPTION
**Pull request by jdrieghe**
### Changed
- `CardSystems` collection class is now a simple array of `CardSystem`s

### Removed
- Dependency on `2dotstwice/collection`

### CODE REVIEW

File: src/UiTPAS/Event/Event/EventCardSystemsUpdatedDeserializer.php [hunk 1]
```diff
@@ -34,7 +33,7 @@ public function deserialize(StringLiteral $data)
             throw new \InvalidArgumentException('Expected cardSystems property to be an array.');
         }
 
-        $cardSystems = new CardSystems();
+        $cardSystems = [];
         foreach ($dto->cardSystems as $cardSystemDTO) {
```
Comment (bertramakers) [inline src/UiTPAS/Event/Event/EventCardSystemsUpdatedDeserializer.php]: This `foreach` can now be changed to `array_map` if I'm not mistaken?

Comment (jdrieghe) [reply src/UiTPAS/Event/Event/EventCardSystemsUpdatedDeserializer.php]: `array_map` to produce an array indexed by specific keys looks pretty messy in my opinion. I'd rather leave it as a `foreach` in that case.

**Comment by jdrieghe** [comment]
@bertramakers I briefly looked into the usages in other packages and I think it can be removed entirely without too much effort.

File: src/UiTPAS/Event/Event/EventCardSystemsUpdatedDeserializer.php (hunk 1 above)
Comment (bertramakers) [reply src/UiTPAS/Event/Event/EventCardSystemsUpdatedDeserializer.php]: Ah I did not notice the keys by id. In that case `foreach` is fine yeah.

**Pull request merged**